src/Card/CardSection/index.js

```diff
@@ -1,123 +1,109 @@
+import CardWrapper from "../components/CardWrapper";
+import { useCard } from "../CardContext";
+import SectionHeader from "./components/SectionHeader";
+import SectionContent from "./components/SectionContent";
 
 import type { Props } from "./index";
 
-const StyledCardSectionIconRight = styled(ChevronDown)`
-  align-self: center;
-  transition: ${({ theme }) => theme.orbit.durationFast};
-`;
-
-StyledCardSectionIconRight.defaultProps = {
-  theme: defaultTheme,
-};
-
-const StyledCardSectionContent = styled.div`
-  font-family: ${({ theme }) => theme.orbit.fontFamily};
-  font-size: ${({ theme }) => theme.orbit.fontSizeTextNormal};
-  line-height: ${({ theme }) => theme.orbit.lineHeightTextNormal};
-  color: ${({ theme }) => theme.orbit.colorTextPrimary};
-  border-top: ${({ theme, expanded }) =>
-    expanded
-      ? `1px solid ${theme.orbit.paletteCloudNormal}`
-      : `0px solid ${theme.orbit.paletteCloudNormal}`};
-  padding-top: ${({ theme, expanded }) => expanded && `${theme.orbit.spaceMedium}`};
-  overflow: hidden;
-`;
-
-StyledCardSectionContent.defaultProps = {
-  theme: defaultTheme,
-};
-
-export const StyledCardSection = styled.div`
-  width: 100%;
-  padding: ${({ theme }) => theme.orbit.spaceMedium};
-  box-sizing: border-box;
-  position: relative;
-  background: ${({ theme }) => theme.orbit.backgroundCard};
-
-  ${media.desktop(css`
-    padding: ${({ theme }) => theme.orbit.spaceLarge};
-  `)}
-`;
-
-StyledCardSection.defaultProps = {
-  theme: defaultTheme,
-};
-
-const StyledCardSectionHeader = styled.div`
-  margin-bottom: ${({ theme, expanded }) => expanded && theme.orbit.spaceMedium};
-  display: flex;
-  flex-direction: row;
-  cursor: pointer;
-  position: relative;
-  min-height: ${({ expandable }) => expandable && getSize(ICON_SIZES.MEDIUM)};
-`;
+const CardSection = ({
+  title,
+  icon,
+  description,
+  children,
+  expandable = false,
+  expanded = false,
+  initialExpanded = false,
+  onClose,
+  header,
+  onExpand,
+  dataTest,
+  actions,
+  noSeparator,
+  dataA11ySection,
+}: Props) => {
+  const [opened, setOpened] = React.useState(initialExpanded || expanded);
+
+  const { onToggle, setExpandedSections, index, roundedBorders, noBorderTop } = useCard();
+
+  React.useEffect(() => {
+    if (expandable && (initialExpanded || expanded)) {
+      setExpandedSections([index]);
+    }
 
-StyledCardSectionHeader.defaultProps = {
-  theme: defaultTheme,
-};
+    if (!initialExpanded) {
+      setOpened(expanded);
+    }
 
-class CardSection extends React.Component<any, Props> {
-  componentDidMount() {
-    const { handleToggleSection, initialExpanded, setInitialExpandedSection } = this.props;
+    return () => {
+      setExpandedSections([]);
+    };
+  }, [expandable, expanded, index, initialExpanded, setExpandedSections]);
 
-    if (initialExpanded) {
-      handleToggleSection();
-      setInitialExpandedSection();
+  const handleClick = () => {
+    if (!expanded) {
```

Comment:
Should be:
```
if(expanded == null)
```
Otherwise produces undesired behaviour when `expanded` is `false`. It's both controlled and uncontrolled component.

Reply:
This is also connected to default value of `expanded`. It believe that it should be set to `undefined` (no default value), because it's than hard to determinate the controlled or uncontrolled variant. Moreover, it seems that there is the bug with double border between let's day `CardHeader` and the first `CardSection`.